src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -130,14 +132,30 @@ void GcsResourceManager::HandleGetAllAvailableResources(
     rpc::GetAllAvailableResourcesReply *reply,
     rpc::SendReplyCallback send_reply_callback) {
   for (const auto &node_resources_entry : cluster_resource_manager_.GetResourceView()) {
-    const auto &node_id = node_resources_entry.first;
-    const auto &node_resources = node_resources_entry.second.GetLocalView();
+    if (node_resources_entry.first == local_node_id_) {
+      continue;
+    }
     rpc::AvailableResources resource;
-    resource.set_node_id(node_id.Binary());
-
+    resource.set_node_id(node_resources_entry.first.Binary());
+    const auto &node_resources = node_resources_entry.second.GetLocalView();
+    const auto node_id = NodeID::FromBinary(node_resources_entry.first.Binary());
+    bool using_resource_reports = RayConfig::instance().gcs_actor_scheduling_enabled() &&
+                                  node_resource_usages_.contains(node_id);
     for (const auto &resource_id : node_resources.available.ResourceIds()) {
-      const auto &resource_value = node_resources.available.Get(resource_id);
       const auto &resource_name = resource_id.Binary();
+      // Because gcs scheduler does not directly update the available resources of
+      // `cluster_resource_manager_`, use the record from resource reports (stored in
+      // `node_resource_usages_`) instead.
```

Comment:
In gcs scheduling, available resources are not updated upon resource reports, because we handle normal_task_resources specifically (see `GcsResourceManager::UpdateFromResourceReport`). But `node_resource_usages_` stores the raw copy of resource reports, so we can use this to get the available resources.